Sourcery/Parsing/Composer.swift

```diff
@@ -98,6 +98,30 @@ struct Composer {
         }
 
         let lookupName = typeName.actualTypeName ?? typeName
+        
+        if lookupName.generic == nil && lookupName.isGeneric {
+            let genericSplit = lookupName.unwrappedTypeName.characters.split(separator: "<")
+            
+            guard genericSplit.count == 2 else {
+                return nil
+            }
+            
+            let typeCharacters = genericSplit[0]
+            var genericCharacters = genericSplit[1]
+            
+            guard genericCharacters.count > 1, genericCharacters.removeLast() == ">" else {
+                return nil
+            }
+            
+            let genericTypes = genericCharacters.split(separator: ",").map { characters in
+                return String(characters).trimmingCharacters(in: [" "])
+            }
+            
+            lookupName.generic = GenericType(name: String(typeCharacters), referencedTypes: genericTypes.flatMap { genericTypeName in
+                return unique[genericTypeName]
```

Comment:
type parameter can be typealias, or another generic, so we should use `resolveType` here, we should test all these cases.